nixos/lib/utils: genJqSecretsReplacementSnippet incorrectly uses `$` to on arithmetic variable, failing shellCheck

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

Nixos modules/services using `genJqSecretsReplacementSnippet` currently fail nixos-rebuilds when the build runs `shellCheck` due to rule `SC2004 (style): $/${} is unnecessary on arithmetic variables`. 

The culprit is how [genJqSecretsReplacementSnippet references inherit_errexit_enabled ](https://github.com/NixOS/nixpkgs/blob/447a511e5f633f3ef4009d3b23d18b60b60b7a1a/nixos/lib/utils.nix#L380). This occurs when running  `nixos-rebuild <options>` with a modules which uses `genJqSecretsReplacementSnippet`

I have only verified this bug for modules [recyclarr](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/misc/recyclarr.nix) and[ netbird/management](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/networking/netbird/management.nix)



### Steps to reproduce

in nixos configuration, set 
`services.netbird.enabled = true;`
or 
`services.recyclarr.enabled = true;` 
or any service which depends on `genJqSecretsReplacementSnippet` 
then try to rebuild via `nixos-rebuild <test/switch/build/...>`

### Expected behaviour

`nixos-rebuild switch ...` successfully builds nixos modules

### Screenshots

[recyclarr](https://github.com/NixOS/nixpkgs/blob/447a511e5f633f3ef4009d3b23d18b60b60b7a1a/nixos/modules/services/misc/recyclarr.nix#L106)
![Image](https://github.com/user-attachments/assets/bb277f3d-220f-4d64-9b71-77799bb29dbc)

[netbird](https://github.com/NixOS/nixpkgs/blob/447a511e5f633f3ef4009d3b23d18b60b60b7a1a/nixos/modules/services/networking/netbird/management.nix#L384)
<img width="1116" alt="Image" src="https://github.com/user-attachments/assets/57d3e850-bbba-4e08-af48-f4132aa9edeb" />

### Relevant log output

```console
In /nix/store/rjk4y760vn4cs9n9mqa0h9m9ra4i4xnv-unit-script-netbird-management-pre-start/bin/netbird-management-pre-start line 19:
(( ! $inherit_errexit_enabled )) && shopt -u inherit_errexit
     ^----------------------^ SC2004 (style): $/${} is unnecessary on arithmetic variables.

For more information:
  https://www.shellcheck.net/wiki/SC2004 -- $/${} is unnecessary on arithmeti...
```

### Additional context

I'm no expert here, so I may be incorrect - `radarr` is a relatively new addition so the issue could be something with my configuration..

Regardless, I suspect adjusting [genJqSecretsReplacementSnippet](https://github.com/NixOS/nixpkgs/blob/447a511e5f633f3ef4009d3b23d18b60b60b7a1a/nixos/lib/utils.nix#L380) `(( ! $inherit_errexit_enabled )) && shopt -u inherit_errexit` --> `(( ! inherit_errexit_enabled )) && shopt -u inherit_errexit` should resolve this error. But, I'm not familiar with the intricacies of shell so that may alter functionality. The suggested code _should_  be fine according to shell's [arithmetic docs](https://www.gnu.org/software/bash/manual/html_node/Shell-Arithmetic.html). I have also modified the function in a local copy of nixpkgs and I can build both modules successfully.  

Wasn't too sure who to notify (could not find a maintainer file for `utils.nix`) so I used git blame. If i pinged the wrong person, sorry in advance

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.12, NixOS, 25.05 (Warbler), 25.05.20250208.a79cfe0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - nixpkgs: `/nix/store/j33wzkzndh41cyyy7i18bqm1srlv84cq-source`

### Notify maintainers

@talyz 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/nixos/lib/utils.nix
+++ b/nixos/lib/utils.nix
@@ -377,7 +377,7 @@ let
          <<'EOF'
         ${toJSON set}
         EOF
-        (( ! $inherit_errexit_enabled )) && shopt -u inherit_errexit
+        (( ! inherit_errexit_enabled )) && shopt -u inherit_errexit
       '';
 
     /*

